zerocopy-derive/src/lib.rs

```diff
@@ -55,6 +55,45 @@ macro_rules! try_or_print {
     };
 }
 
+/// Inline size assertion annotations
+/// Any non-generic type can be check if the size is as expected or not in compile time
+/// # Implementation
+/// ```
+/// # use zerocopy_derive::inline_assert_size_eq;
+/// # use static_assertions;
+/// #[inline_assert_size_eq(4)]
+/// struct MyStruct {
+///     val: i32
+/// }
+/// ```
+/// # Example that will fail
+/// ```compile_fail
+/// # use zero_derive::inline_assert_size_eq;
+/// # use static_assertions;
+/// #[inline_assert_size_eq(1)]
+/// struct MyStruct {
+///     val: i32
+/// }
+/// ```
+#[proc_macro_attribute]
+pub fn inline_assert_size_eq(
+    args: proc_macro::TokenStream,
+    input: proc_macro::TokenStream,
+) -> proc_macro::TokenStream {
+    let ast: Result<DeriveInput, _> = syn::parse(input.clone());
+    let expected_size: Result<usize, _> = args.to_string().trim().parse();
+    let mut ret: proc_macro::TokenStream = input;
+    if let (Ok(ast), Ok(size)) = (ast, expected_size) {
+        let name = &ast.ident;
+        ret = quote! {
+            #ast
+            static_assertions::const_assert!(core::mem::size_of::<#name>() == #size);
```

Comment:
We can't defer to the `static_assertions` crate, since we don't want that to be a public dependency of zerocopy, nor do we want to be an explicit dependency of our consumers.

You should instead add a hygeinic helper macro to [`macro_util.rs`](https://github.com/google/zerocopy/blob/main/src/macro_util.rs) and invoke that in the proc macro.

Reply:
Done